rust/shalom_dart_codegen/templates/operation.dart.jinja

```diff
@@ -1,91 +1,95 @@
-{% macro base_object_impl(typename, selections) %}
-
-/// class memberes
-{% for selection in selections %}
-final {{type_name_for_selection(selection)}} {{selection.selection_name}};
-
-{% endfor %}
-
-// keywordargs constructor
-{{typename}}({
+{% macro generate_class(typename, selections, root, is_root=false) %}
+/// {{typename}} class with selected fields from query
+{% set classname = 'Request' ~ typename %}
+class {{classname}} {
+  /// Class fields
+  {% for selection in selections %}
+  {% if selection.kind == "Object" %}
+  final Request{{root}}{{selection.selection_name|capitalize}}? {{selection.selection_name}};
+  {% else %}
+  final {{type_name_for_selection(selection)}} {{selection.selection_name}};
+  {% endif %}
+  {% endfor %}
+
+  /// Constructor
+  {{classname}}({
     {% for selection in selections %}
-    required this.{{selection.selection_name}},
+    {% if not selection.is_optional %}required {% endif %}this.{{selection.selection_name}},
     {% endfor %}
-});
-
+  });
 
-static {{typename}} fromJson(Map<String, dynamic> data) {
+  /// Creates from JSON
+  factory {{classname}}.fromJson(Map<String, dynamic> json) => {{classname}}(
     {% for selection in selections %}
-    {% if selection.kind == "Scalar" %}
-        final {{ selection.selection_name }}_value = data['{{ selection.selection_name }}'] as {{ type_name_for_selection(selection) }};
-
-        {% endif %}
-    {% endfor %}
-
-  return {{typename}}(
-    {% for selection in selections %}
-    {{selection.selection_name}}: {{selection.selection_name}}_value,
+    {{selection.selection_name}}: {% if selection.kind == "Object" %}
+      json['{{selection.selection_name}}'] != null 
+        ? Request{{root}}{{selection.selection_name|capitalize}}.fromJson(json['{{selection.selection_name}}'] as Map<String, dynamic>)
+        : null
+    {% else %}
+      json['{{selection.selection_name}}'] as {{type_name_for_selection(selection)}}
+    {% endif %},
     {% endfor %}
+  );
+
+  /// Updates from JSON
+  {{classname}} updateWithJson(Map<String, dynamic> data) {
```

Comment:
if you modify anything, you should  in the same lines as the previous thing. hard to review like this.
please move all declarations to their original LOC